eng/doc/fips/UserGuide.md

```diff
@@ -639,11 +900,330 @@ It internally uses sha512.New() to compute the checksum.
+
+**Implementation**
+
+<details><summary>OpenSSL</summary>
+
+`sig` is verified using using [EVP_PKEY_verify] with `RSA_PKCS1_PSS_PADDING` pad mode.
+
+</details>
+
+<details><summary>CNG</summary>
+
+`sig` is verified using [BCryptVerifySignature] with [PSS_PADDING_INFO] padding information and `BCRYPT_PAD_PSS` pad mode.
+
+</details>
+
+#### func [GenerateKey](https://pkg.go.dev/crypto/rsa#GenerateKey)
+
+```go
+func rsa.GenerateKey(rand io.Reader, bits int) (priv *rsa.PrivateKey, err error)
+```
+
+GenerateKey generates a public and private key pair.
+
+**Requirements**
+
+- `rand` must be boring.RandReader. `crypto/rand.Reader` normally meet this invariant as it is assigned to boring.RandReader in the crypto/rand init function.
+- `bits` must be either 2048 or 3072.
+
+**Implementation**
+
+<details><summary>OpenSSL</summary>
+
+`priv` is a wrapper around [EVP_PKEY] generated using [EVP_PKEY_keygen].
+
+</details>
+
+<details><summary>CNG</summary>
+
+`priv` is generated using [BCryptGenerateKeyPair] with the [algorithm identifier] `BCRYPT_RSA_ALGORITHM`.
+
+</details>
+
+#### func [GenerateMultiPrimeKey](https://pkg.go.dev/crypto/rsa#GenerateMultiPrimeKey)
+
+```go
+func rsa.GenerateMultiPrimeKey(rand io.Reader, nprimes int, bits int) (priv *rsa.PrivateKey, err error)
+```
+
+GenerateMultiPrimeKey generates a multi-prime RSA keypair of the given bit size.
+
+**Requirements**
+
+- `rand` must be boring.RandReader. `crypto/rand.Reader` normally meet this invariant as it is assigned to boring.RandReader in the crypto/rand init function.
+- `nprimes` must be 2. 
+- `bits` must be either 2048 or 3072.
+
+**Implementation**
+
+<details><summary>OpenSSL</summary>
+
+`priv` is a wrapper around [EVP_PKEY] generated using [EVP_PKEY_keygen].
+
+</details>
+
+<details><summary>CNG</summary>
+
+`priv` is generated using [BCryptGenerateKeyPair] with the [algorithm identifier] `BCRYPT_RSA_ALGORITHM`.
+
+</details>
+
+#### func [PrivateKey.Decrypt](https://pkg.go.dev/crypto/rsa#PrivateKey.Decrypt)
+
+```go
+func (priv *PrivateKey) Decrypt(rand io.Reader, ciphertext []byte, opts crypto.DecrypterOpts) (plaintext []byte, err error)
+```
+
+Decrypt decrypts `ciphertext` with `priv`.
+
+The depcrypt function depends on `opts`:
+
+- If `opts` is nil, it calls [rsa.DecryptPKCS1v15](#func-decryptpkcs1v15)`(rand, priv, ciphertext)`.
+- If `opts` type is `*rsa.OAEPOptions`, it calls [rsa.DecryptOAEP](#func-decryptoaep)`(opts.Hash.New(), rand, priv, ciphertext, opts.Label)`.
+- If `opts` type is `*rsa.PKCS1v15DecryptOptions` and `opts.SessionKeyLen > 0`, it calls [rsa.DecryptPKCS1v15SessionKey](#func-decryptpkcs1v15sessionkey)`(rand, priv, ciphertext, plaintext)` with a random `plaintext`.
+- If `opts` type is `*rsa.PKCS1v15DecryptOptions` and `opts.SessionKeyLen == 0`, it calls [rsa.DecryptPKCS1v15](#func-decryptpkcs1v15)`(rand, priv, ciphertext)`.
+- Else it returns an error.
+
+Check the function named in the applicable case for the parameters' restrictions.

```

Comment:
I know I suggested this wording, 😄 but reading it again and maybe understanding it a bit better, I think it should be more specific that this is an action you need to take to make sure you're using the func right:

```suggestion
Each case may impose additional parameter requirements. After determining which case applies, check the linked function to find the additional restrictions.
```